libgraphics/source/DrawCallResolver.cxx

```diff
@@ -36,18 +36,29 @@ void DrawCallResolver::destroy()
     if (descriptorSetLayout) base_ref->get().device.destroyDescriptorSetLayout(descriptorSetLayout);
 }
 
-void DrawCallResolver::prepareForDraw(const std::vector<std::reference_wrapper<const RenderObject>> &sceneInformation,
-                                      const CameraData &camera, const uint32_t frameIndex)
+void DrawCallResolver::prepareForDraw(std::vector<std::reference_wrapper<const RenderObject>> &sceneInformation,
+                                      const uint32_t frameIndex)
 {
     auto &frame = frames.at(frameIndex);
-    std::vector<DrawBatch> packedDraws;
+    frame.packedDraws.clear();
+    frame.pipelineBatch.clear();
     std::vector<gpu_object::UniformBufferObject> objectGPUData;
-    uint32_t drawCount = 0;
+    std::uint32_t drawCount = 0;
 
+    std::sort(sceneInformation.begin(), sceneInformation.end(), [](const auto &first, const auto &second) {
+        return first.get().pipelineID == second.get().pipelineID && first.get().meshID == second.get().meshID;
```

Comment:
You return true when the scenes are equal, whereas it should return true when the first is lower than the second.

https://en.cppreference.com/w/cpp/algorithm/sort

> comp | - | comparison function object (i.e. an object that satisfies the requirements of Compare) which returns ​true if the first argument is less than (i.e. is ordered before) the second.
comp 	- 	comparison function object (i.e. an object that satisfies the requirements of [Compare](https://en.cppreference.com/w/cpp/named_req/Compare)) which returns ​true if the first argument is less than (i.e. is ordered before) the second.

And anyway, why do you need to perform a sort on each frame ?

Reply:
I need to order the object first by pipeline ID, then by mesh ID (optional, useful for the future). I want to minimize changing pipeline as much as possible.

Reply:
Ok, but your comparison function is wrong. It should return true if the first object is less than the second.

Reply:
But there is no lesser/greater, I only want to group equivalent objects.